sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -87,16 +87,20 @@ public void openConnectionIfNeeded() {
             return;
         }
 
+        Uri addressUri = this.addressUris.stream().findFirst().get();
+
         // connection state listener submits an open connection task for an endpoint
-        // and only submits the task for that endpoint when the previous task has been
-        // completed
+        // and only submits the task for that endpoint if and only if the previous task has been
+        // completed and the endpoint is used by a container which was part of the connection
+        // warm up flow
         // it is okay to lose a task, since each task will attempt to attain a certain no. of
         // connection as denoted by the min required no. of connections for that endpoint
-        if (this.endpointValidationInProgress.compareAndSet(false, true)) {
+        if (this.endpointValidationInProgress.compareAndSet(false, true)
+                && this.proactiveOpenConnectionsProcessor.isAddressUriUnderOpenConnectionsFlow(addressUri.getURIAsString())) {
```

Comment:
I think the order of conditions is wrong

You set endpointValidationInProgress even when not under openConnectionFow - and the flag would never be reset in that case.

probably better to just add below to exit early

if (!this.proactiveOpenConnectionsProcessor.isAddressUriUnderOpenConnectionsFlow(addressUri.getURIAsString()) {
    return;
}